src/App.tsx

```diff
@@ -34,62 +35,110 @@ export function getReorderedGoods(
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
 
-      <ul>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reverseGoods = () => {
+    this.setState((state) => ({ isReversed: !state.isReversed }));
+  };
+
+  resetChange = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const sortedGoods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info',
+              { 'is-light': sortType !== SortType.ALPABET })}
+            onClick={this.sortByAlphabet}
+          >
+            Sort alphabetically
+          </button>
+          <button
+            type="button"
+            className={classNames('button is-success',
+              { 'is-light': sortType !== SortType.LENGTH })}
+            onClick={this.sortByLength}
+          >
+            Sort by length
+          </button>
+          <button
+            type="button"
+            className={classNames('button is-warning',
+              { 'is-light': !isReversed })}
+            onClick={this.reverseGoods}
+          >
+            Reverse
+          </button>
+
+          {(sortType !== SortType.NONE || isReversed)
+            ? (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={this.resetChange}
+              >
+                Reset
+              </button>
+            )
+            : null}
+
+        </div>
+
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {sortedGoods.map(good => (
+            <li data-cy="Good" key={good}>
```

Comment:
have had reject while use a name from array, mentor typed that it can be not unique